### PR TITLE
Implemented Cross Entropy algorithm

### PR DESCRIPTION
**Pull request by Egomeh**
This adds an implementation of the Cross Entropy algorithm.

Implemented as part of a bachelors thesis by.

### CODE REVIEW

**Comment by Ulfgard** [comment]
Hi,

thanks for the pull request. CrossEntropy is a bad name because it collides with another class name in shark: ObjectiveFunctions/Loss/CrossEntropy.h

Further, a bit of documentation (e.g. reference to a paper and a short description of the algorithm) would be nice

**Comment by Egomeh** [comment]
Hi Oswin

Is it okay to call it "CrossEntropyMethod", perheps with a reference to
http://www.maths.uq.edu.au/~kroese/ps/aortut.pdf

And, sorry I'm not used to contribute to these kinds of projects, but where
would I add the documentation?

Best regards

Jens Holm

**Comment by Ulfgard** [comment]
Thank you for your contribution! I just merged your branch and it is now part of the next Shark release!
Yay!
